sqlmesh/core/model/definition.py

```diff
@@ -552,19 +552,6 @@ def update_schema(
                     tuple(str(part) for part in table.parts),
                     {k: v.sql(dialect=self.dialect) for k, v in mapping_schema.items()},
                 )
-            else:
-                # Reset the entire mapping if at least one upstream dependency is missing from the mapping
-                # to prevent partial mappings from being used.
-                if self.contains_star_projection:
-                    logger.warning(
-                        "Missing schema for model '%s' referenced in model '%s'. Run `sqlmesh create_external_models` "
-                        "and / or make sure that the model '%s' can be rendered at parse time",
-                        dep,
-                        self.name,
-                        dep,
-                    )
-                self.mapping_schema.clear()
-                return
```

Comment:
The reasoning / assumption I made for getting rid of this section was that we already optimize @ load time anyway, so we will warn about missing schemas there and hence not needed to warn here as well earlier.

Regarding @izeigerman's [comment](https://github.com/TobikoData/sqlmesh/pull/1675#discussion_r1385780345), I don't think we want to omit warnings even for edge models, because we can't disambiguate between a valid, but external, model reference and an invalid one (e.g. that contains a typo). Happy to continue this discussion though as I may be missing context, though to my understanding this PR also doesn't significantly change the previous behaviour as we would already warn for these models @ optimize time.

Reply:
Btw `self.mapping_schema` is involved in the computation of the metadata hash. Does this mean if we get rid of this section we'd also need to make a new migration script?

Reply:
yea, so if you upgrade sqlglot minor, it should be fine

Reply:
For reference, the current behavior in main w.r.t. the edge models argument (with both selecting everything & a single column w/ explicit type and alias):

```
➜  cat models/test.sql
model (name test, kind full);

select * from external
➜  sqlmesh render test
Missing schema for model 'external' referenced in model 'test'. Run `sqlmesh create_external_models` and / or make sure that the model 'external' can be rendered at parse time
Query cannot be optimized due to missing schema for model 'external'. Run `sqlmesh create_external_models` and / or make sure that the model 'test' can be rendered at parse time
Query cannot be optimized due to missing schema for model 'external'. Run `sqlmesh create_external_models` and / or make sure that the model 'test' can be rendered at parse time
SELECT
  *
FROM "external" AS "external"
➜  cat models/test.sql
model (name test, kind full);

select a::int as a from external
➜  sqlmesh render test
Query cannot be optimized due to missing schema for model 'external'. Run `sqlmesh create_external_models` and / or make sure that the model 'test' can be rendered at parse time
Query cannot be optimized due to missing schema for model 'external'. Run `sqlmesh create_external_models` and / or make sure that the model 'test' can be rendered at parse time
SELECT
  CAST("a" AS INT) AS "a"
FROM "external" AS "external"
```

Whereas, this PR changes to:
```
➜  cat models/test.sql
model (name test, kind full);

select * from external
➜  sqlmesh render test
Query cannot be optimized due to missing schema for model 'external'. Run `sqlmesh create_external_models` and / or make sure that the model 'test' can be rendered at parse time.
Query cannot be optimized due to missing schema for model 'external'. Run `sqlmesh create_external_models` and / or make sure that the model 'test' can be rendered at parse time.
SELECT
  *
FROM "external" AS "external"
➜  cat models/test.sql
model (name test, kind full);

select a::int as a from external
➜  sqlmesh render test
Query cannot be optimized due to missing schema for model 'external'. Run `sqlmesh create_external_models` and / or make sure that the model 'test' can be rendered at parse time.
Query cannot be optimized due to missing schema for model 'external'. Run `sqlmesh create_external_models` and / or make sure that the model 'test' can be rendered at parse time.
SELECT
  CAST("a" AS INT) AS "a"
FROM "external" AS "external"
```